jetbrains-core/tst/software/aws/toolkits/jetbrains/utils/rules/RunWithRealCredentials.kt

```diff
@@ -0,0 +1,88 @@
+import software.aws.toolkits.jetbrains.core.MockClientManager
+import software.aws.toolkits.jetbrains.core.credentials.AwsConnectionManager
+import software.aws.toolkits.jetbrains.core.credentials.CredentialManager
+import software.aws.toolkits.jetbrains.core.credentials.DefaultAwsConnectionManager
+import software.aws.toolkits.jetbrains.core.credentials.DefaultCredentialManager
+import software.aws.toolkits.jetbrains.core.credentials.profiles.DEFAULT_PROFILE_ID
+import software.aws.toolkits.jetbrains.core.credentials.profiles.ProfileCredentialProviderFactory
+import software.aws.toolkits.jetbrains.core.credentials.waitUntilConnectionStateIsStable
+import software.aws.toolkits.jetbrains.core.region.AwsRegionProvider
+
+class RunWithRealCredentials : TestRule {
+    /**
+     * Marks a unit test, or unit test class, as requiring AWS credentials to run. Will always use the default AWS profile, and by default us-west-2
+     */
+    annotation class RequiresRealCredentials
+
+    private val projectSupplier: () -> Project
+
+    constructor(projectRule: ProjectRule) : super() {
+        this.projectSupplier = { projectRule.project }
+    }
+
+    constructor(projectRule: CodeInsightTestFixtureRule) : super() {
+        this.projectSupplier = { projectRule.project }
+    }
+
+    lateinit var disposable: Disposable
+    override fun apply(base: Statement, description: Description): Statement = if (description.getAnnotation(RequiresRealCredentials::class.java) == null &&
+        description.testClass.getAnnotation(RequiresRealCredentials::class.java) == null
+    ) {
+        base
+    } else {
+        object : Statement() {
+            override fun evaluate() {
+                val disposable = Disposer.newDisposable()
+                try {
+                    val project = projectSupplier.invoke()
+
+                    MockClientManager.useRealImplementations(disposable)
+
+                    // ProfileCredentialProviderFactory is designed to be a singleton, so replace it with a new instance
+                    ExtensionTestUtil.maskExtensions(DefaultCredentialManager.EP_NAME, listOf(ProfileCredentialProviderFactory()), disposable)
+
+                    ApplicationManager.getApplication().replaceService(CredentialManager::class.java, DefaultCredentialManager(), disposable)
+                    ApplicationManager.getApplication().replaceService(AwsResourceCache::class.java, DefaultAwsResourceCache(), disposable)
+                    project.replaceService(AwsConnectionManager::class.java, DefaultAwsConnectionManager(project), disposable)
+
+                    val credentialIdentifier = CredentialManager.getInstance().getCredentialIdentifierById(DEFAULT_PROFILE_ID)
+                        ?: throw IllegalStateException("RunWithRealCredentials requires a default AWS profile!")
+                    val region = AwsRegionProvider.getInstance()["us-west-2"] ?: throw IllegalStateException("Can't locate us-west-2")
```

Comment:
probably should be:

`getEnv("AWS_DEFAULT_REGION") ?: us-west-2 ?: throw`

Reply:
Ah, fair

Reply:
Fixed